dotCMS/src/main/java/com/dotmarketing/business/cache/provider/CacheSizingUtil.java

```diff
@@ -0,0 +1,257 @@
+
+        if (object == null) {
+            return 0;
+        }
+
+        if (!(object instanceof Serializable)) {
+            Logger.warn(this.getClass(), object.getClass() + " not serializable: " + object);
+            return -1;
+        }
+
+        if (object instanceof String) {
+            return ((String) object).length() * 8;
+
+        }
+
+        byteOutputStream.reset();
+        try (ObjectOutputStream objectOutputStream = new ObjectOutputStream(byteOutputStream)) {
+            objectOutputStream.writeObject(object);
+            objectOutputStream.flush();
+            objectOutputStream.close();
+
+            return byteOutputStream.toByteArray().length;
+        } catch (Exception e) {
+            Logger.warnAndDebug(this.getClass(), "failed writing:" + object.getClass() + " : " + e.getMessage(), e);
+            return 0;
+        }
+
+    }
+
+    /**
+     * returns the size of an objects class, based on the classes properties
+     * 
+     * @param object
+     * @return
+     */
+    long sizeOfClass(Object object) {
+
+        return ClassLayout.parseClass(object.getClass()).instanceSize();
+    }
+
+    public static boolean useCompressedOops = true;
+
+    public static int retainedSize(Object obj) {
+        return retainedSize(obj, new HashMap<>());
+    }
+
+    @SuppressWarnings("restriction")
+    private static int retainedSize(Object obj, HashMap<Object, Object> calculated) {
+        Object ref = null;
+        try {
+            if (obj == null)
+                throw new NullPointerException();
+            calculated.put(obj, obj);
+            Class<?> cls = obj.getClass();
+            if (cls.isArray()) {
+                int arraysize = UnsafeAccess.UNSAFE.arrayBaseOffset(cls)
+                                + UnsafeAccess.UNSAFE.arrayIndexScale(cls) * Array.getLength(obj);
+                if (!cls.getComponentType().isPrimitive()) {
+                    Object[] arr = (Object[]) obj;
+                    for (Object comp : arr) {
+                        if (comp != null && !isCalculated(calculated, comp))
+                            arraysize += retainedSize(comp, calculated);
+                    }
+                }
+                return arraysize;
+            } else {
+                int objectsize = sizeof(cls);
+                for (Field f : getAllNonStaticFields(obj.getClass())) {
+                    Class<?> fcls = f.getType();
+                    if (fcls.isPrimitive())
+                        continue;
+                    f.setAccessible(true);
+                    ref = f.get(obj);
+                    if (ref != null && !isCalculated(calculated, ref)) {
+                        calculated.put(ref, ref);
+                        int referentSize = retainedSize(ref, calculated);
+                        objectsize += referentSize;
+                    }
+                }
+                return objectsize;
+            }
+        } catch (Throwable e) {
```

Comment:
Codacy found an issue: [A catch statement should never catch throwable since it includes errors.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6315945)